source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1054 @@
+                                 * this to correctly place the variables later on.
+                                 */
+                                double index = indx+nx*ys;
+
+                                temporary_variables[0].push_back(vertex(dim-1));
+                                temporary_variables[1].push_back(index-1);
+
+                                for (unsigned int i=0; i<dim; ++i)
+                                  temporary_variables[i+2].push_back(vel[corner][i]*year_in_seconds);
+                              }
+                          }
+
+                        if (dim == 3)
+                          {
+                            double indy = 2+vertex(1)/dy;
+                            if (indy - floor(indy) >= precision)
+                              continue;
+
+                            double index = (indy-1)*nx+indx;
+
+                            temporary_variables[1].push_back(index-1);
+                            temporary_variables[0].push_back(vertex(dim-1));   //z component
+
+                            for (unsigned int i=0; i<dim; ++i)
+                              temporary_variables[i+2].push_back(vel[corner][i]*year_in_seconds);
+                          }
+                      }
+                  }
+
+          //Run fastscape on single processor.
+          if (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0)
+            {
+              TimerOutput::Scope timer_section(this->get_computing_timer(), "Fastscape 1 proc");
+              /*
+               * Initialize the variables that will be sent to fastscape.
+               * These have to be doubles of array_size, which C++ doesn't like,
+               * so they're initialized this way.
+               */
+              std::unique_ptr<double[]> h (new double[array_size]);
+              std::unique_ptr<double[]> vx (new double[array_size]);
+              std::unique_ptr<double[]> vy (new double[array_size]);
+              std::unique_ptr<double[]> vz (new double[array_size]);
+              std::unique_ptr<double[]> kf (new double[array_size]);
+              std::unique_ptr<double[]> kd (new double[array_size]);
+              std::unique_ptr<double[]> slopep (new double[array_size]);
+              int istep = 0;
+              int steps = nstep;
+              std::srand(fs_seed);
+              std::vector<double> h_old(array_size);
+
+              //Create variables for output directory and restart file
+              std::string filename;
+              filename = this->get_output_directory();
+              const char *c=filename.c_str();
+              int length = filename.length();
+              const std::string restart_filename = filename + "fastscape_h_restart.txt";
+              const std::string restart_step_filename = filename + "fastscape_steps_restart.txt";
+
+              //Initialize all FastScape variables.
```

Comment:
This is odd, so the way I had it doesn't initialize them to zero. I found that by changing it from
 
`std::unique_ptr<double[]> h (new double[array_size]);`   
to
`std::unique_ptr<double[]> h (new double[array_size]());`   

It will initialize everything except the very last element to zero, and the last element seems to have a random number. I can initialize them as array_size+1, and then all the important ones are set to zero, and fastscape doesn't seem to care if the array is too large.